tabcmd/commands/datasources_and_workbooks/export_command.py

```diff
@@ -65,51 +68,85 @@ def run_command(args):
         session = Session()
         server = session.create_session(args)
         logger.info("===== Requesting '{0}' from the server...".format(args.url))
-        try:
-            if args.fullpdf:  # it's a workbook
-                workbook = ExportCommand.get_content_url_for_workbook(args.url)
-                workbook_from_list = ExportCommand.get_wb_by_content_url(logger, server, workbook)
+        view_content_url, wb_content_url = ExportCommand.parse_export_url_to_workbook_and_view(args.url)
+        logger.debug("Parsed to {}, {}".format(wb_content_url, view_content_url))
+        if args.fullpdf:  # it's a workbook
+            try:
+                workbook_item = ExportCommand.get_wb_by_content_url(logger, server, wb_content_url)
+            except Exception as e:
+                Errors.exit_with_error(logger, "Error getting workbook info from server", e)
+
+            try:
                 req_option_pdf = TSC.PDFRequestOptions(maxage=1)
-                server.workbooks.populate_pdf(workbook_from_list, req_option_pdf)
-                output = workbook_from_list.pdf
-                default_filename = "{}.pdf".format(workbook)
+                server.workbooks.populate_pdf(workbook_item, req_option_pdf)
+            except Exception as e:
+                Errors.exit_with_error(logger, "Error downloading workbook pdf", e)
+            output = workbook_item.pdf
+            default_filename = "{}.pdf".format(wb_content_url)
+
+        elif args.pdf or args.png or args.csv:  # it's a view
+            try:
+                view_item = ExportCommand.get_view_by_content_url(logger, server, view_content_url)
+            except Exception as e:
+                Errors.exit_with_error(logger, "Error getting view info from server", e)
+
+            try:
+                if args.pdf:
+                    output = ExportCommand.download_pdf(server, view_item)
+                    default_filename = "{}.pdf".format(view_item.name)
+                elif args.csv:
+                    output = ExportCommand.download_csv(server, view_item)
+                    default_filename = "{}.csv".format(view_item.name)
+                elif args.png:
+                    output = ExportCommand.download_png(server, view_item)
+                    default_filename = "{}.png".format(view_item.name)
+            except Exception as e:
+                Errors.exit_with_error(logger, "Error downloading view from server", e)
```

Comment:
Minor - Can this be a bit more descriptive? - for example  "Please choose from png/csv/pdf" .

Reply:
At this spot we have confirmed that the arguments were correct and we ran into an error when trying to actually download the view.